### PR TITLE
Refactored Quotas to support firebase implementation

### PR DESCRIPTION
**Pull request by adshayanB**
- Updated Quotas to support firebase implementation
- Quotas can be found using their marketid as it is only a 1:1 mapping
- Handles when a market does not have a quota
- Ability to view/update a quota harnessing firestore our backend db 

It is to note that markets are still hard coded, so hence the quotas for all the markets are currently the same. This will be changed through the next cr where each farm will have individual markets with individual quotas

Update flow (adding soumil to the quota):

image
image
image

### CODE REVIEW

**Review by PreyanshK — APPROVED** [review]
Good work with the refactoring

**Review by RishanR — CHANGES_REQUESTED** [review]
Instead of adding a map called saleAmount to the Quota class, you can just extend ProduceQutoa class to have the saleAmount. So every quota has a sale amount. I think this will also reflect the db a little better, where every produce is a map of goal and sale. In this case, every ProduceQuota has a goal and sale attribute.

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt [hunk 1]
```diff
@@ -18,13 +20,18 @@ class QuotasRepository {
 
     data class Quota(
         val id: String,
-        val produceQuotaList : List<ProduceQuota>,
+        val produceQuotaList: List<ProduceQuota>,
+        val saleAmount: MutableMap<String, Int>
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:24]: Basically I mean to remove this saleAmount map, and instead just add a produceSaleAmount variable in ProduceQuota which is just of type Int.

**Review by SoumilDharaskar — APPROVED** [review]
Great work! LGTM!

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt [hunk 2]
```diff
@@ -1,115 +1,78 @@
...
-        }.flowOn(Dispatchers.IO)
-    }
+    suspend fun getQuota(id: String): Quota? {
+        val docRef = db.collection("quotas").document(id)
 
-    fun getQuota(id : String): Quota? {
-        return quotasList.firstOrNull { quota -> quota.id == id }
-    }
+        val quotas = docRef.get().await()?.data?.get("produce")
+        val saleCount = docRef.get().await()?.data?.get("sale")
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:39]: do docRef.get().await()? once and store it in a variable. You're doing two document reads here.

Comment (adshayanB) [reply app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:39]: I need to pull both dicts right? One for sale and one for produce

Comment (adshayanB) [reply app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:39]: This is how the collection looks like:
image

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt [hunk 3]
```diff
@@ -1,115 +1,78 @@
...
+        for (prod in produce) {
+            produceQuota[prod.produceName] = prod.produceGoalAmount
+        }
+
+        val data = hashMapOf(
+            "produce" to produceQuota,
+            "sale" to SALES,
+        )
+        db.collection("quotas").document(market.id)
+            .set(data)
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt]: You actually don't need to reset the document, you can just update the produce map. You can change all this code to just the following:

```
db.collection("quotas").document(market.id)
    .update("produce", produce.associate(
        it.produceName to it.produceGoalAmount
    ))
```

Comment (RishanR) [reply app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt]: Make sure this still works but it should be good. And then you can just add the onSucceessListner and stuff on top of it,

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt [hunk 4]
```diff
@@ -1,115 +1,80 @@
...
+                val produceQuota = ProduceQuota(
+                    produceName = key,
+                    produceGoalAmount = value,
+                    saleAmount = sale.getOrDefault(key, 0)
+                )
+                produceList.add(produceQuota)
+            }
+            return Quota(
+                id = id,
+                produceQuotaList = produceList
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt:57]: you can change this to:
```
return Quota(
    id = id,
    produceQuotaList = quotasMap.map{ (produceName, goal) ->
        ProduceQuota(
            produceName = produceName,
            produceGoalAmount = goal,
            saleAmount = sale.getOrDefault(produceName, 0),
        )
    }
}
```

File: app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt (hunk 3 above)
Comment (adshayanB) [reply app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt]: Yup tested it it works :)

**Review by RishanR — APPROVED** [review]
Looking good!

**Pull request merged**